lib/withOnyx.js

```diff
@@ -67,12 +78,33 @@ export default function (mapOnyxToState) {
                 });
             }
 
+            /**
+             * This method is used externally by sendDataToConnection to prevent unnecessary renders while a component
+             * still in a loading state. The temporary initial state is saved to the component instance and setState()
+             * only called once all the necessary data has been collected.
+             *
+             * @param {String} statePropertyName
+             * @param {*} val
+             */
+            setInitialState(statePropertyName, val) {
+                this.tempState[statePropertyName] = val;
```

Comment:
I think this method should throw an Error or at least print a warning and skip execution if it's called after `loading` is set to `false`